src/modules/0.0.16/aws_account/index.ts

```diff
@@ -1,50 +1,49 @@
 import { AWS, } from '../../../services/aws_macros'
 import { AwsAccountEntity, } from './entity'
-import { Context, Crud2, Mapper2, Module2, } from '../../interfaces'
-import * as metadata from './module.json'
+import { Context, Crud2, Mapper2, ModuleBase, } from '../../interfaces'
+import * as metadata from './module.json' // TODO: Eliminate this?
 
-export const AwsAccount: Module2 = new Module2({
-  ...metadata,
-  provides: {
-    context: {
-      // This function is `async function () {` instead of `async () => {` because that enables the
-      // `this` keyword within the function based on the objec it is being called from, so the
-      // `getAwsClient` function can access the correct `orm` object with the appropriate creds and
-      // read out the right AWS creds and create an AWS client also attached to the current context,
-      // which will be different for different users. WARNING: Explicitly trying to access via
-      // `AwsAccount.provides.context.getAwsClient` would instead use the context *template* that is
-      // global to the codebase.
-      async getAwsClient() {
-        const orm = this.orm;
-        if (this.awsClient) return this.awsClient;
-        const awsCreds = await orm.findOne(AwsAccount.mappers.awsAccount.entity);
-        this.awsClient = new AWS({
-          region: awsCreds.region,
-          credentials: {
-            accessKeyId: awsCreds.accessKeyId,
-            secretAccessKey: awsCreds.secretAccessKey,
-          },
-        });
-        return this.awsClient;
-      },
-      awsClient: null, // Just reserving this name to guard against collisions between modules.
+class AwsAccount extends ModuleBase {
+  constructor() { super(); super.init(); }
+  dirname = __dirname;
+  dependencies = metadata.dependencies;
```

Comment:
with this we could technically remove the metadata.json or are we still using it somewhere else?

Reply:
So we still use it, along with the `dependencies` field, in the migration generation logic in the `src/scripts` directory, and I can't think of a good way to eliminate it, because we explicitly *don't* want these module `index.ts` files to be loaded since they try to auto-load the migration files that may or may not exist and then blow up in weird ways.